### PR TITLE
[WIP] Switching syntax styles option

### PR DESCRIPTION
**Pull request by mariacamilaremolinagutierrez**
Now there is an option to change to a different syntax style that opens a new tab in the console with the selected color scheme:

image1

### CODE REVIEW

**Comment by mariacamilaremolinagutierrez** [comment]
@ccordoba12 @minrk I was doing this PR in order to find out if there was a way of changing the syntax_style without creating a new tab but that instead it got updated automatically in the same shell I've been working on.

The initial intention was inspired by spyder's https://github.com/spyder-ide/spyder/pull/3908. However I was unable to do so. Do any of you know if this is possible? Or do I have to restart it and initialize it with the new config (as in this PR)?

**Comment by takluyver** [comment]
`JupyterWidget` has a `_syntax_style_changed()` method, which should be called automatically when `syntax_style` is set. So if you can get the active JupyterWidget instance, you should be able to do:

```python
jupyter_widget.syntax_style = new_style
```

It might be that that doesn't work, though, if no-one's tested it before.

File: qtconsole/mainwindow.py [hunk 1]
```diff
@@ -730,6 +739,20 @@ def toggleFullScreen(self):
     def set_paging_active_frontend(self, paging):
         self.active_frontend._set_paging(paging)
 
+    def get_available_syntax_styles(self):
```
Comment (takluyver) [inline qtconsole/mainwindow.py]: We can use a pygments function to find the available styles:
http://pygments.org/docs/styles/#getting-a-list-of-available-styles

**Comment by wmvanvliet** [comment]
awesome!

File: qtconsole/mainwindow.py (hunk 1 above)
Comment (mariacamilaremolinagutierrez) [inline qtconsole/mainwindow.py]: Oh this is way better! Thank you!

**Comment by mariacamilaremolinagutierrez** [comment]
> JupyterWidget has a ```_syntax_style_changed()``` method, which should be called automatically when ```syntax_style``` is set. So if you can get the active JupyterWidget instance, you should be able to do: ```jupyter_widget.syntax_style = new_style```

@takluyver I tried to do so before in another project (Spyder IDE) where I have the JupyterWidget instance: 

image

The interesting thing was that the style_sheet was updated automatically (without restarting it or making a new tab). But the syntax style did not, it only got updated after I restarted Spyder. 

This makes me think about your last line and that it maybe doesn't work. I tried to check both functions a bit but I couldn't figure out fast what could be wrong with the ```_syntax_style_changed()``` one.

**Comment by wmvanvliet** [comment]
Since this creates a new kernel, shouldn't this option do in the "Kernel" menu? Is there no way to change the pygments coloring of the currently running kernel? That would be super.

**Comment by ccordoba12** [comment]
@wmvanvliet, we'll fix that because we need it for Spyder.

**Comment by ccordoba12** [comment]
Closing in favor of #217.

**Pull request closed without merging**